omas/tests/test_omas_core.py

```diff
@@ -127,6 +127,22 @@ def test_misc(self):
         )
         return
 
+    def test_data_slicing(self):
+        ods = ODS()
+        ods['langmuir_probes.embedded.0.name'] = '1'
+        ods['langmuir_probes.embedded.1.name'] = '12'
+        ods['langmuir_probes.embedded.2.name'] = '123'
+        assert ods['langmuir_probes']['embedded.:.name'][2] == '123'
+
+        ods = ODS(consistency_check=False)
+        for k in range(3):
+            ods[f'langmuir_probes.embedded.{k}.name'] = [float(k)] * (k + 1)
```

Comment:
Doesn't assigning this into 'name' force it to be a string? Maybe put it into ...embedded.{k}.time or ...embedded.{k}.saturation_current_ion.data or something instead? I can't even make this work in the OMFIT command box because it wants a string (`ValueError: langmuir_probes.embedded.1.name is of type <class 'numpy.ndarray'> must be a scalar of type STR_0D`).

Reply:
@eldond notice that I disabled the IMAS consistency checks with `consistency_check=False`

Reply:
Okay. I should've seen that.